### PR TITLE
Newsletter

### PR DESCRIPTION
**Pull request by jcaru614**

### CODE REVIEW

File: src/pages/api/newsletter/subscribe.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  try {
+    const { email } = await request.json();
+
+    if (!email || !email.includes('@')) {
+      return NextResponse.json(
+        { error: 'Please provide a valid email address' },
+        { status: 400 },
+      );
+    }
+
```
Comment (jcaru614) [inline src/pages/api/newsletter/subscribe.ts:14]: We may wanna check for duplicate emails:

const existingSubscriber = await prisma.contacts.findUnique(email);

    if (existingSubscriber) {
      return status 409 you are subscribed already
    }

File: src/components/EmailCollector.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+
+const EmailCollector: React.FC = () => {
```
Comment (jcaru614) [inline src/components/EmailCollector.tsx:3]: At some point move next to the search bar, perhaps? Not sure if it looks too crowded up there though.

File: src/pages/api/newsletter/subscribe.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import prisma from '@/lib/prisma';
+import { NextResponse } from 'next/server';
+
+export async function POST(request: Request) {
```
Comment (jcaru614) [inline src/pages/api/newsletter/subscribe.ts:4]: We will probably need an unsubscribe functionality as well

Comment (ahmadanees) [reply src/pages/api/newsletter/subscribe.ts]: i think we should implement that at the bottom of the newsletter email

**Pull request merged**